File: task1/task1b/ComputerGame.cpp
#include "ComputerGame.h"
#include <iostream>

ComputerGame::ComputerGame(){}

ComputerGame::ComputerGame(int id, const string& name, double price, const string& category, int volume, const string& pub): Product(id, name, price, category, volume), gamePublisher(pub) {}

void ComputerGame::display_product_info(){
    Product::display_product_info();
    cout << "Game Publisher: " << getPublisher() << endl;
}

// Setter
void ComputerGame::setPublisher(const string& pub){
    gamePublisher = pub;
}

string ComputerGame::getPublisher() const{
    return gamePublisher;
}

File: task3/task3a_b/src/main/java/com/task3a_b/ComputerGame.java
package com.task3a_b;

public class ComputerGame extends Product{
    private String gamePublisher;

    public ComputerGame(){

    }

    public ComputerGame(int id, String name, double price, String category, int volume, String pub){
        super(id, name, price, category, volume);
        gamePublisher = pub;
    }

    @Override
    public void display_product_info(){
        super.display_product_info();
        System.out.println("Game Publisher: "+getPublisher());
    }

    // Setter
    public void setPublisher(String pub){
        gamePublisher = pub;
    }

    // Getter
    public String getPublisher(){
        return gamePublisher;
    }
}
